mixnet/protocol/src/lib.rs

```diff
@@ -41,21 +36,28 @@ impl<'a> Body<'a> {
         }
     }
 
-    async fn read_sphinx_packet<R>(reader: &mut R) -> Result<Body<'a>, Box<dyn Error>>
+    async fn read_sphinx_packet<R>(reader: &mut R) -> Result<Body, Box<dyn Error>>
     where
         R: AsyncRead + Unpin,
     {
-        let mut buf = Vec::new();
-        reader.read_to_end(&mut buf).await?;
+        let size = reader.read_u64().await?;
+        let mut buf = vec![0; size as usize];
+        reader.read_exact(&mut buf).await?;
+
```

Comment:
Now, we use the lengh-prefixed protocol, because `ClientNotifier` holds a connection with the mixclient and uses it to send multiple payloads.

In near future, we also need to introduce reusing connections between mixnodes.